plotfunctions.py

```diff
@@ -288,7 +288,12 @@ def plotDomain(self, ax=None, verbose=False, plotskip=1):
                     offsets =[float(x) for x in allpdict[p+'.offsets'].split()]
                 else:
                     offsets = [0.0]
-                offsetnormal = np.array(allpdict[p+'.normal'])
+                # GRY: Using try - except here to start. Is the default value None if the field does not exists though?
+                try:
+                    offsetnormal = np.array(allpdict[p+'.offset_vector'])
+                except:
+                    print("offset_vector not specified. Using legacy interface: .normal")
+                    offsetnormal = np.array(allpdict[p+'.normal'])
```

Comment:
@lawrenceccheung , is there a default value if keys to not exist in allpdict? If so, I can revert this to an if else statement

Reply:
I believe the default value for items which were not specified by the user is what's given the `config.yaml` `default` section, and if there's nothing there, then it's just `None`.

Reply:
Got it, OK, the default for normal and offset_vector is [0,0,0]. In that case, this "try, except" block should be correct. For new setups, offset_vector will be read in for plotting/validating inputs. However, if it doesn't exist, normal will be read instead. For the actual case setups, both offset_vector and normal will be written to maintain compatibility with either amr-wind builds.